tests/visualization_tests/matplotlib_tests/test_optimization_history.py

```diff
@@ -83,11 +83,17 @@ def objective(trial: Trial) -> float:
     figure = plot_optimization_history(studies)
     assert len(figure.get_lines()) == n_studies
 
+    legend_texts = []
+    expected_legend_texts = []
     for i, legend in enumerate(figure.legend().get_texts()):
+        legend_texts.append(legend.get_text())
+
         if i < n_studies:
-            assert legend.get_text() == f"Best Values of {studies[i].study_name}"
+            expected_legend_texts.append(f"Best Values of {studies[i].study_name}")
         else:
-            assert legend.get_text() == f"Objective Value of {studies[i-n_studies].study_name}"
+            expected_legend_texts.append(f"Objective Value of {studies[i-n_studies].study_name}")
+
+    assert sorted(legend_texts) == sorted(expected_legend_texts)
```

Comment:
I don't think it is a good practice to create expected values using the actual values.
For example, if `figure.legend().get_texts()` returns smaller number of items than `n_studies`, I guess the assertion still passes.

How about preparing the expected values based on `n_studies`?

The following is just an example.
```suggestion
    expected_legend_texts = []
    for i in range(n_studies):
        expected_legend_texts.append(f"Best Values of {studies[i].study_name}")
        expected_legend_texts.append(f"Objective Value of {studies[i].study_name}")
    legend_texts = [legend.get_text() for legend in figure.legend().get_texts()]
    assert sorted(legend_texts) == sorted(expected_legend_texts)
```